### PR TITLE
IGAPP-1021: Language change with selected poi broken

### PR DESCRIPTION
**Pull request by f1sh1918**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

http://localhost:9000/testumgebung/es/locations?name=berlin
Now switch the language
The poi should stay selected and the url correct

### CODE REVIEW

File: api-client/src/routes/pathname.ts [hunk 1]
```diff
@@ -55,8 +55,7 @@ export const pathnameFromRouteInformation = (routeInformation: NonNullableRouteI
   if (
     routeInformation.route === DASHBOARD_ROUTE ||
     routeInformation.route === CATEGORIES_ROUTE ||
-    routeInformation.route === EVENTS_ROUTE ||
-    routeInformation.route === POIS_ROUTE
+    routeInformation.route === EVENTS_ROUTE
   ) {
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply api-client/src/routes/pathname.ts:59]: But urlSLug only exist for POIS_ROUTE,
therefore i need a separat `if (routeInformation.route === POIS_ROUTE )` condition before or the code will break.
I have to revert your suggestions so..

File: api-client/src/routes/pathname.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ import {
   CITY_NOT_COOPERATING_ROUTE,
   SHELTER_ROUTE
 } from '.'
-import { NonNullableRouteInformationType } from '..'
+import { locationName, NonNullableRouteInformationType } from '..'
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply api-client/src/routes/pathname.ts:16]: ok i renamed it to `nameQueryParam`

But the if condition as mentioned above is necessary  for the urlSlug. It has nothing to do with the queryParamName.

File: api-client/src/routes/pathname.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (steffenkleinle) [reply api-client/src/routes/pathname.ts:59]: Okay I see, its not for dashboard/categories. However, there should be a separate PoisRouteInformationType then, makes no sense for events.

File: api-client/src/routes/pathname.ts [hunk 3]
```diff
@@ -65,6 +64,15 @@ export const pathnameFromRouteInformation = (routeInformation: NonNullableRouteI
...
     return constructPathname([routeInformation.cityCode, routeInformation.languageCode, routeInformation.route])
   }
+  if (routeInformation.route === POIS_ROUTE) {
+    const { cityCode, languageCode, route, urlSlug } = routeInformation
+    if (urlSlug) {
+      //  https://integreat.app/augsburg/de/locations?name=tuer-an-tuer
+      return `${constructPathname([cityCode, languageCode, route])}?${nameQueryParam}=${urlSlug}`
+    }
+    //  https://integreat.app/augsburg/de/locations
+    return constructPathname([cityCode, languageCode, route])
```
Comment (steffenkleinle) [inline api-client/src/routes/pathname.ts:74]:
```suggestion
    const pathname = constructPathname([cityCode, languageCode, route])
    //  https://integreat.app/augsburg/de/locations, https://integreat.app/augsburg/de/locations?name=tuer-an-tuer
    return urlSlug ? `${pathname}?${nameQueryParam}=${urlSlug}` : pathname
```

**Review by steffenkleinle — APPROVED** [review]
Sorry for the bad suggestion, I thought dashboard and categories are handled separately.

File: web/src/routes/PoisPage.tsx [hunk 4]
```diff
@@ -106,18 +106,16 @@ const PoisPage = ({ cityCode, languageCode, cityModel, pathname, languages }: Ci
...
-      name,
-      code
-    }
-  })
+  const languageChangePaths = languages.map(({ code, name }) => ({
+    path: pathnameFromRouteInformation({
+      route: POIS_ROUTE,
+      cityCode,
+      languageCode: code,
+      urlSlug: poi?.urlSlug
```
[the remark this answers is not in the document]
Comment (f1sh1918) [reply web/src/routes/PoisPage.tsx:114]: yes

File: api-client/src/routes/pathname.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (f1sh1918) [reply api-client/src/routes/pathname.ts:59]: [IGAPP-1037](https://issues.tuerantuer.org/browse/IGAPP-1037)

**Review by SteffiStuffel — APPROVED** [review]
Looks good. Tested on Chrome. 

When moving the app and then changing the language it is zoomed again to the POI, which is not the behavior i would expect as a user.

**Comment by steffenkleinle** [comment]
Not sure how that could be avoided, we'd probably had to save the current zoom/coordinates somewhere in the url. I think zooming in to the selected poi is better then starting over with the default bounding box

**Pull request merged**